spring-cloud-gcp-vision/src/main/java/org/springframework/cloud/gcp/vision/DocumentOcrTemplate.java

```diff
@@ -0,0 +1,250 @@
+				.setType(Type.DOCUMENT_TEXT_DETECTION)
+				.build();
+
+		AsyncAnnotateFileRequest request = AsyncAnnotateFileRequest.newBuilder()
+				.addFeatures(feature)
+				.setInputConfig(inputConfig)
+				.setOutputConfig(outputConfig)
+				.build();
+
+		OperationFuture<AsyncBatchAnnotateFilesResponse, OperationMetadata> result =
+				imageAnnotatorClient.asyncBatchAnnotateFilesAsync(Collections.singletonList(request));
+
+		return extractOcrResultFuture(result);
+	}
+
+	/**
+	 * Parses the OCR output files with the specified {@code jsonFilesetPrefix}. This method
+	 * assumes that all of the OCR output files with the prefix are a part of the same
+	 * document. This method is useful for processing a collection of output files produced by
+	 * the same document.
+	 *
+	 * @param jsonFilePathPrefix the folder location containing all of the JSON files of OCR
+	 *     output
+	 * @return A {@link DocumentOcrResultSet} describing the OCR content of a document
+	 */
+	public DocumentOcrResultSet parseOcrOutputFileSet(GoogleStorageLocation jsonFilePathPrefix) {
+		String nonNullPrefix = (jsonFilePathPrefix.getBlobName() == null) ? "" : jsonFilePathPrefix.getBlobName();
+
+		Page<Blob> blobsInFolder = this.storage.list(
+				jsonFilePathPrefix.getBucketName(),
+				BlobListOption.currentDirectory(),
+				BlobListOption.prefix(nonNullPrefix));
+
+		List<Blob> blobPages =
+				StreamSupport.stream(blobsInFolder.getValues().spliterator(), false)
+						.filter(blob -> blob.getContentType().equals("application/octet-stream"))
+						.sorted(Comparator.comparingInt(blob -> extractPageNumber(blob)))
+						.collect(Collectors.toList());
+
+		return new DocumentOcrResultSet(blobPages, this.storage);
+	}
+
+	/**
+	 * Parses a single JSON output file and returns the parsed OCR content.
+	 *
+	 * @param jsonFile the location of the JSON output file
+	 * @return the {@link TextAnnotation} containing the OCR results
+	 * @throws InvalidProtocolBufferException if the JSON file cannot be deserialized into a
+	 *     {@link TextAnnotation} object
+	 */
+	public TextAnnotation parseOcrOutputFile(GoogleStorageLocation jsonFile)
+			throws InvalidProtocolBufferException {
+		if (!jsonFile.isFile()) {
+			throw new IllegalArgumentException(
+					"Provided jsonOutputFile location is not a valid file location: " + jsonFile);
+		}
+
+		Blob jsonOutputBlob = this.storage.get(
+				BlobId.of(jsonFile.getBucketName(), jsonFile.getBlobName()));
+		return parseJsonBlob(jsonOutputBlob);
+	}
+
+	static TextAnnotation parseJsonBlob(Blob blob) throws InvalidProtocolBufferException {
+		AnnotateFileResponse.Builder annotateFileResponseBuilder = AnnotateFileResponse.newBuilder();
+		String jsonContent = new String(blob.getContent());
+		JsonFormat.parser().merge(jsonContent, annotateFileResponseBuilder);
+
+		AnnotateFileResponse annotateFileResponse = annotateFileResponseBuilder.build();
+
+		return annotateFileResponse.getResponses(0).getFullTextAnnotation();
```

Comment:
Is it possible to have more than 1 response?

Reply:
Nope, this is also safe for the same reason as above. Only our code will be able to provide the response object (with 1 response inside).